src/mumble/Log.cpp

```diff
@@ -415,20 +415,26 @@ QString Log::validHtml(const QString &html, bool allowReplacement, QTextCursor *
 
 	qtd.adjustSize();
 	QSizeF s = qtd.size();
+	int messageSize = s.width() * s.height();
+	int allowedSize = 2048 * 2048;
+
+	if (messageSize > allowedSize || !s.isValid()) {
+		QString errorSizeMessage = tr("[[ Text object too large to display ]]");
+		if (tc) {
+			tc->insertText(errorSizeMessage);
+			return QString();
+		} else {
+			return errorSizeMessage;
+		}
+	}
 
-	if (!valid || !s.isValid() || (s.width() > qr.width()) || (s.height() > qr.height())) {
-		qtd.setPlainText(html);
-		qtd.adjustSize();
-		s = qtd.size();
-
-		if (!s.isValid() || (s.width() > qr.width()) || (s.height() > qr.height())) {
-			QString errorMessage = tr("[[ Text object too large to display ]]");
-			if (tc) {
-				tc->insertText(errorMessage);
-				return QString();
-			} else {
-				return errorMessage;
-			}
+	if (!valid) {
```

Comment:
I would move this above `qtd.adjustSize()`, etc. -- `valid` is already set there, so we don't need to wait to use it until we're down here.